### PR TITLE
Working Demo

### PR DESCRIPTION
**Pull request by willscott**
The last chunk of work revolved around a read returning a bucket, which contains multiple data items. The response handler now attempts decryption of each individually.

The Consistency proof of concept binary shows that we can write and subsequently read an item.

### CODE REVIEW

File: cuckoo/table.go [hunk 1]
```diff
@@ -38,19 +38,29 @@ func (i *Item) Equals(other *Item) bool {
 	}
 	return i.Bucket1 == other.Bucket1 &&
 		i.Bucket2 == other.Bucket2 &&
-		i.Id == other.Id;
+		i.Id == other.Id
+}
+
+// Bucket returns the bucket in a table that the Item is in, if it is in the table.
+func (i *Item) Bucket(table *Table) int {
```
Comment (ryscheng) [inline cuckoo/table.go]: Looks like this is unused? 
I'd remove this if unused. If used, seems more appropriate as a table.Find() => returns bucket number or something like that.

File: libtalek/client.go [hunk 2]
```diff
@@ -68,37 +55,63 @@ func NewClient(name string, config ClientConfig, leader common.LeaderInterface)
...
+	config := c.config.Load().(ClientConfig)
+	return config.DataSize
 }
 
+// Publish a new message to the end of a topic.
 func (c *Client) Publish(handle *Topic, data []byte) error {
 	config := c.config.Load().(ClientConfig)
-	write_args, err := handle.GeneratePublish(config.CommonConfig, data)
+
+	if len(data) > config.DataSize-PublishingOverhead {
```
Comment (ryscheng) [inline libtalek/client.go]: Wouldn't it make more sense to have DataSize be the maximum size of any user-published data? As opposed to making the client keep track of what this overhead is.

Comment (willscott) [inline libtalek/client.go]: DataSize is the item size the server uses right now. libTalek should probably expose an 'ItemSize' property that the user can see that is config.DataSize-PublishingOverhead?

Comment (ryscheng) [inline libtalek/client.go]: Makes sense

File: cuckoo/table.go (hunk 1 above)
Comment (willscott) [inline cuckoo/table.go]: added test.

**Comment by ryscheng** [comment]
👍

**Pull request merged**